kratos/utilities/force_and_torque_utils.cpp

```diff
@@ -0,0 +1,123 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:        BSD License
+//	                Kratos default license: kratos/license.txt
+//
+//  Main Authors:   Máté Kelemen
+//
+
+/* System includes */
+
+/* External includes */
+
+/* Project includes */
+#include "utilities/force_and_torque_utils.h"
+#include "utilities/parallel_utilities.h"
+#include "utilities/reduction_utilities.h"
+#include "utilities/math_utils.h"
+
+namespace Kratos
+{
+
+
+array_1d<double,3> ForceAndTorqueUtils::SumForce(
+    const ModelPart& rModelPart,
+    const Variable<array_1d<double,3>>& rForceVariable)
+{
+    KRATOS_TRY
+
+    array_1d<double,3> force = block_for_each<SumReduction<array_1d<double,3>>>(
+        rModelPart.GetCommunicator().LocalMesh().Nodes(),
+        [&rForceVariable](const Node<3>& rNode) -> array_1d<double,3>
+        {
+            return rNode.GetSolutionStepValue(rForceVariable);
+        }
+    );
+
+    force = rModelPart.GetCommunicator().GetDataCommunicator().SumAll(force);
+
+    return force;
+
+    KRATOS_CATCH("")
+}
+
+
+std::array<array_1d<double,3>,2> ForceAndTorqueUtils::SumForceAndTorque(
+    const ModelPart& rModelPart,
+    const array_1d<double,3>& rReferencePoint,
+    const Variable<array_1d<double,3>>& rForceVariable,
+    const Variable<array_1d<double,3>>& rTorqueVariable)
+{
+    KRATOS_TRY
+
+    using CompoundArray = std::array<array_1d<double,3>,2>;
+    using CompoundTuple = std::tuple<array_1d<double,3>,array_1d<double,3>>;
+    using CompoundReduction = CombinedReduction<
+        SumReduction<array_1d<double,3>>,
+        SumReduction<array_1d<double,3>>
+    >;
+
+    std::function<CompoundTuple(const Node<3>&)> compute_force_and_moment;
+
+    // Check whether nodes have the torque variable and set the lambda accordingly.
+    // Note: multiple node types within the model part are not supported.
+    //       In that case, this check would have to performed for each node.
+    if (rModelPart.GetCommunicator().LocalMesh().Nodes().begin()->SolutionStepsDataHas(rTorqueVariable)) {
```

Comment:
I'm strongly against this check. If the user provides the point but the variable is in the database, you'll be computing the total torque from the provided variable, not using the provided point, meaning that the current behavior is way different to the one expected.

Again I insist on having two methods. One with point and one without. Either with same name (method overload) or with different one as @philbucher suggested.

Reply:
Right, I get what you mean now. The original intent was to compute the total reaction force and moment of a model part, but that info was lost along the way and is not reflected in the function's name anymore.

I'll modify this one such that only sums up the variables. What should the intended function's name be (```ComputeReactionForceAndMoment``` for example)?

Reply:
What I'd like to have as a user is:

1. `SumForce`: calculates the resultant of the given variable :heavy_check_mark: (we already have it)
2. `SumForceAndTorque`: calculates the resultant force and moment from the two given force and moment variables. Long story short, this does the reduction of the two variables. The signature would be something like
```c++
    std::array<array_1d<double,3>,2> SumForceAndTorque(
        const ModelPart& rModelPart,
        const Variable<array_1d<double,3>>& rForceVariable,
        const Variable<array_1d<double,3>>& rTorqueVariable);
```
3. `SumForceAndTorqueFromReferencePoint`: calculates the resultant force and moment from the given force variable. For the resultant moment, in here we accumulate the cross product.
```c++
    std::array<array_1d<double,3>,2> SumForceAndTorqueFromReferencePoint(
        const ModelPart& rModelPart,
        const Variable<array_1d<double,3>>& rForceVariable,
        const array_1d<double,3>& rReferencePoint);
```
(I think this is what you had in mind too).

Reply:
More or less yes, but I'd sum up the moment variable as well in addition to the cross product in 3. As it stands right now, getting the total moment exerted by the model's reactions can only be computed by making two function calls, looping through every node twice. Preventing that is the main goal of this utility.

Reply:
> More or less yes, but I'd sum up the moment variable as well in addition to the cross product in 3. As it stands right now, getting the total moment exerted by the model's reactions can only be computed by making two function calls, looping through every node twice. Preventing that is the main goal of this utility.

Exactly: it would be great to avoid looping over nodes twice!

Reply:
Then I'd add a 4th method. The point since adding the reaction moment only makes sense in presence of rotational DOFs which is not the CFD case.